esp/services/ws_fileio/ws_fileioservice.cpp

```diff
@@ -30,15 +30,45 @@
 
 void CWsFileIOEx::init(IPropertyTree *cfg, const char *process, const char *service)
 {
+    CTpWrapper tpWrapper;
+    tpWrapper.getTpDropZones(9999, nullptr, false, allTpDropZones); //version 9999: get the latest information about dropzone
 }
 
-bool CWsFileIOEx::CheckServerAccess(const char* targetDZNameOrAddress, const char* relPath, StringBuffer& netAddr, StringBuffer& absPath)
+bool CWsFileIOEx::CheckServerAccess(const char* targetDZNameOrAddress, const char* netAddrReq, const char* relPath, StringBuffer& netAddr, StringBuffer& absPath)
 {
     if (!targetDZNameOrAddress || (targetDZNameOrAddress[0] == 0) || !relPath || (relPath[0] == 0))
         return false;
 
 #ifdef _CONTAINERIZED
-    UNIMPLEMENTED_X("CONTAINERIZED(CWsFileIOEx::CheckServerAccess)");
+    bool isIp4Req = isIPAddress(netAddrReq);
+    ForEachItemIn(i, allTpDropZones)
+    {
+        IConstTpDropZone& dropZone = allTpDropZones.item(i);
+        if (!dropZone.getECLWatchVisible())
+            continue;
+
+        const char* name = dropZone.getName();
+        if (isEmptyString(name) || !streq(targetDZNameOrAddress, name))
+            continue;
+
+        const char* prefix = dropZone.getPath();
+        if (isEmptyString(prefix))
+            continue;
+
+        IArrayOf<IConstTpMachine>& tpMachines = dropZone.getTpMachines();
+        ForEachItemIn(ii, tpMachines)
+        {
+            IConstTpMachine& tpMachine = tpMachines.item(ii);
+            if (!isEmptyString(netAddrReq) && !matchNetAddressRequest(netAddrReq, isIp4Req, tpMachine))
+                continue;
+
+            netAddr.set(tpMachine.getNetaddress());
+            absPath.set(prefix);
+            addPathSepChar(absPath);
+            absPath.append(relPath);
+            return true;
+        }
+    }
 #else
     netAddr.clear();
```

Comment:
Should this code also check netAddrReq?

Reply:
Could the container code also be used for bare-metal?

Reply:
> Should this code also check netAddrReq?
A user may use the targetDZNameOrAddress to pass in a network address request.  

> Could the container code also be used for bare-metal?
The container code does not handle the legacy structure for bare-metal.

Reply:
But the same request should work, whether or not it is targeting a bare metal or k8s system.  I suspect in most cases the name will be passed an no ip, so not a big issue.

Why doesn't the container code handle the legacy structure for baremetal?  What would go wrong if the container code was always used.

Reply:
The legacy structure is referred to the line 100. In many settings for bare metal, a DropZone setting does not have the ServerList section. But, the DropZone setting has an '@computer' attribute. Based on that attribute, the network address of the DropZone can be found from the Hardware section in the environment.xml. If the container code is used for those bare metal settings, the network address of the DropZone cannot be found.